cmd/kubeadm/app/apis/kubeadm/v1beta4/types.go

```diff
@@ -443,3 +443,38 @@ type Patches struct {
 	// +optional
 	Directory string `json:"directory,omitempty"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ResetConfiguration contains a list of elements that is specific "kubeadm reset"-only runtime information.
+type ResetConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// SkipPhases is a list of phases to skip during command execution.
+	// The list of phases can be obtained with the "kubeadm reset phase --help" command.
+	// The flag "--skip-phases" takes precedence over this field.
+	// +optional
+	SkipPhases []string `json:"skipPhases,omitempty"`
+
+	// ForceReset flag instructs kubeadm to reset the node without prompting for confirmation
+	// The flag "--force" takes precedence over this field.
+	// +optional
+	ForceReset bool `json:"forceReset,omitempty"`
+
+	// CertificatesDir specifies where to store or look for all required certificates.
+	// Cannot mix the config for both the resetConfig and flag to make it consistent with kubeadm init.
+	// +optional
+	CertificatesDir string `json:"certificatesDir,omitempty"`
```

Comment:
Similar to `CRISocket`, this can be retrieved from `ClusterConfiguration.CertificatesDir` if the apiserver is accessible.

Reply:
same reason as above, 

```
kubeadm reset --help

Flags:
      --cert-dir string                   The path to the directory where the certificates are stored. If specified, clean this directory. (default "/etc/kubernetes/pki")

```

Reply:
ideally CertificatesDir  should not be in ClusterConfiguration. it should be a node specific config.
then kubeadm can store differently the PKI directory on different nodes.

maybe we can work on that for API v1.
we could store a custom structure in the Node object under an annotation, or write in named config maps...to be decided.

right now, to overwrite the clusterconfig value, one can use the --cert-dir flag per command execution.

but it makes sense to me to have it in ResetConfiguration as well.
again, precedence should be:

1. flag, if specified
2. local config, if specified
3. cluster config, if there is connection
4. autodetect if all else fails / is missing
5. just fallback to some default constant

of course, we don't do this in all places correctly, but we can consider this the complete precedence list.

Reply:
Done